Swift/ApiDateFormattingService.swift.twig

```diff
@@ -1,23 +1,31 @@
-import LeadKit
-import ObjectMapper
+{%- import 'macroses/common.utils.twig' as utils -%}
+{%- import 'macroses/class.utils.twig' as classUtils -%}
 
-final class ApiDateFormattingService: DateFormattingService {
+import LeadKit
+import SwiftDate
 
-    static let shared = ApiDateFormattingService()
+enum ApiDateFormat: String, LeadKit.DateFormat, Singleton {
 
-    private init() {
-        super.init(formattingArguments: [
-            {% for dateFormat in dateFormats -%}
-                DateFormattingArguments(dateFormat: "{{ dateFormat }}"){% if not (loop.last) %},{%- endif %}
-            {% endfor %}
-        ])
+    {% for format in dateFormats -%}
+    {%- set classType = classUtils.classType(format.type) -%}
+    case {{ utils.dateFormatName(format.fieldName, classType) }} = "{{ format.dateFormat }}"
+    {% endfor %}
+    var swiftDateFormat: SwiftDate.DateFormat {
+        return SwiftDate.DateFormat.custom(rawValue)
     }
+
 }
 
-extension ApiDateFormattingService {
+final class ApiDateFormattingService: DateFormattingService {
+
+    typealias DateFormatType = ApiDateFormat
+
+    var currentRegion: Region = .Local()
 
-    func mappingTransform(for dateFormat: String) -> TransformOf<Date, String> {
-        return mappingTransform(for: DateFormattingArguments(dateFormat: dateFormat))
+    static let shared = ApiDateFormattingService()
+
+    private init() {
```

Comment:
what for?

Reply:
private or init?)
Give me example where you need to subclass this service. Take in account that generated code will use ApiDateFormattingService, not any subclasses

Reply:
so `private init() {}` ?

Reply:
done